Update documentation to routes, move all route HTTPResponses to API file.

SynthesizedSoul opened the pull request:
The motivation for this commit is to address some potential confusion
to newcomers on how this Python application behaves.

I've moved and updated the HTTPResponse code for the
/ping, /start, /move, and /end endpoints to the API file.

Additionally, I've added documentation to the routing to make it easier
to jump in and understand each endpoints purpose.

SynthesizedSoul commented on app/main.py:
```diff
@@ -59,4 +73,5 @@ def ping():
         host=os.getenv('IP', '0.0.0.0'),
         port=os.getenv('PORT', '8080'),
-        debug=True)
+        debug=os.getenv('DEBUG', True)
```
Not sure why this wasn't an env before, will be useful to allow it to be toggle by the env.

joram replied:
nice :+1:

SynthesizedSoul commented on app/main.py:
```diff
@@ -1,55 +1,69 @@
+    """
+    TODO: If you intend to have a stateful snake AI,
+            initialize your snake state here using the
+            requests data if necessary.
```
Should be `request's`.

joram replied:
`requests` is the name of a library... but I don't thing we use it here... you're correct with the comma

SynthesizedSoul commented on app/main.py:
```diff
@@ -1,55 +1,69 @@
-
-@bottle.route('/')
-def static():
-    return "the server is running"
```
This felt a bit unnecessary to me. I guess you could argue it's an easy way to check if the server is alive but I also don't like the plaintext response. 

It was also throwing a linting error for me because it conflicts with the other function definition below.

joram replied:
we now have the ping endpoint as a sanity check the server is up, so it doesn't need to exist anymore, good call

brandonb927 replied:
This endpoint is used in most of the Zero-to-snake tutorials and exists in all, if not most, snake clients. I agree the function name should change, however the endpoint should remain.

SynthesizedSoul replied:
Ahh, this is good to know. Will look at the tutorials and update as appropriate.

Thanks!

joram commented on app/main.py:
```diff
@@ -1,55 +1,69 @@
 
 @bottle.post('/end')
 def end():
-    data = bottle.request.json
```
I like it in to be consistent with other starter snakes.

SynthesizedSoul replied:
Totally, I'll get it back in here!

joram replied:
other than this one thing looks great. approved !

SynthesizedSoul commented:
@joram @brandonb927 I've updated the PR based on the feedback.

I addressed the issue of the `/` route by using Bottle's SimpleEngine templating language to provide a better landing page for first time users to direct them towards the documentation and move away from plaintext responses.

If it's creating too much bloat I'll revert that commit and return to the original plaintext version of the endpoint but with the updated function definition.

brandonb927 commented:
@SynthesizedSoul I'm going to add a ticket to the roadmap to return an HTML document similar to what was introduced here in the root of all the existing starter snakes. 🍻
